Shared/Samples/Find route around barriers/FindRouteAroundBarriersView.Model.swift

```diff
@@ -0,0 +1,207 @@
+        init() {
+            routeGraphicsOverlay.addGraphics([routeGraphic, directionGraphic])
+            graphicsOverlays = [routeGraphicsOverlay, barrierGraphicsOverlay, stopGraphicsOverlay]
+            updateRouteInfoText()
+        }
+        
+        // MARK: Methods
+        
+        /// Adds a stop graphic to the map.
+        /// - Parameter point: The point to add the stop graphic at.
+        func addStopGraphic(at point: Point) {
+            // Create a text symbol with the next index.
+            let textSymbol = TextSymbol(
+                text: "\(stopsCount + 1)",
+                color: .white,
+                size: 20,
+                horizontalAlignment: .center,
+                verticalAlignment: .middle
+            )
+            textSymbol.offsetY = stopSymbol.offsetY
+            
+            // Create a graphic with the marker symbol and text symbol.
+            let compositeSymbol = CompositeSymbol(symbols: [stopSymbol, textSymbol])
+            let stopGraphic = Graphic(geometry: point, symbol: compositeSymbol)
+            
+            // Add the new graphic to the stop graphics overlay.
+            stopGraphicsOverlay.addGraphic(stopGraphic)
+            stopsCount += 1
+        }
+        
+        /// Adds a barrier graphic to the map.
+        /// - Parameter point: The point to add the barrier graphic at.
+        func addBarrierGraphic(at point: Point) {
+            // Buffer the point and create the barrier symbol.
+            let bufferedGeometry = GeometryEngine.buffer(around: point, distance: 500)
+            let barrierSymbol = SimpleFillSymbol(style: .diagonalCross, color: .red)
+            
+            // Create a graphic from the symbol and buffer and add it to the graphics overlay.
+            let barrierGraphic = Graphic(geometry: bufferedGeometry, symbol: barrierSymbol)
+            barrierGraphicsOverlay.addGraphic(barrierGraphic)
+        }
+        
+        /// Resets all the features on the map associated with a given feature type.
+        /// - Parameter featureType: The feature type to remove from the map.
+        func reset(featureType: AddableFeature) {
+            if featureType == .stop {
+                // Reset the stops.
+                stopGraphicsOverlay.removeAllGraphics()
+                stopsCount = 0
+                
+                // Reset the route.
+                route = nil
+                routeGraphic.geometry = nil
+                directionGraphic.geometry = nil
+                updateRouteInfoText()
+            } else {
+                barrierGraphicsOverlay.removeAllGraphics()
+            }
+        }
+        
+        /// Routes using the route parameters and the current stops and barriers on the map.
+        func route() async throws {
+            guard let routeParameters, stopsCount >= 2 else { return }
+            
+            // Update the route parameters' stops using the stop graphics.
+            var stops = [Stop]()
```

Comment:
Do you need this and the `barriers` array? Can't you just append the stop or barrier to the parameters after you create an instance?

Reply:
To my knowledge, there isn't a way to append stops to the route parameter since set methods overwrite the previous values I think. I removed clear methods since, if that's the case, they are not necessary.